File: C/exo4_3_pere.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
#include <string.h>

int main()
{
    int tube[2];
    if(pipe(tube)!=0)
    {
        printf("Erreur lors de la création du tube.\n");
        exit(1);
    }
    
    pid_t pid_fils=fork();
    if (pid_fils==0)
    {
        char fd[10];
        sprintf(fd, "%d", tube[0]);  //metre le descripteur en string
        execl("./programme3_fils", "programme3_fils", fd, NULL);//il faut compiler le programme fils pour avoir l'exécutable "programme_fils"
    }
    else
    {
        close(tube[0]);
        char *message="Message bien reçu par le fils.";
        write(tube[1], message, strlen(message)+1);  //+1 pour \0 à la fin de la chaine
        wait(NULL);
    }
    return 0;
}

File: python/exo4_3_pere.py
import os
import sys

tube_r, tube_w = os.pipe()

if tube_r is None :
        print("Erreur lors de la création du tube.")
        sys.exit(1)

pid_fils = os.fork()

if pid_fils == 0 :
        fd = str(tube_r)    #metre le descripteur en string
        os.execl("./programme3_fils.py", "programme3_fils.py", fd)  #il faut avoir le programme fils "programme3_fils.py"
else :
        os.close(tube_r)
        message = "Message bien reçu par le fils."
        os.write(tube_w, (message + "\0").encode())     #+1 pour \0 à la fin de la chaine
        os.wait()